server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetMappingAction.java

```diff
@@ -97,37 +92,25 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         final HttpChannel httpChannel = request.getHttpChannel();
         return channel -> new RestCancellableNodeClient(client, httpChannel).admin()
             .indices()
-            .getMappings(
-                getMappingsRequest,
-                new DispatchingRestToXContentListener<>(threadPool.executor(ThreadPool.Names.MANAGEMENT), channel, request).map(
-                    getMappingsResponse -> new RestGetMappingsResponse(getMappingsResponse, threadPool::relativeTimeInMillis, timeout)
-                )
-            );
-    }
-
-    private static final class RestGetMappingsResponse implements ToXContentObject {
-        private final GetMappingsResponse response;
-        private final LongSupplier relativeTimeSupplierMillis;
-        private final TimeValue timeout;
-        private final long startTimeMs;
-
-        private RestGetMappingsResponse(GetMappingsResponse response, LongSupplier relativeTimeSupplierMillis, TimeValue timeout) {
-            this.response = response;
-            this.relativeTimeSupplierMillis = relativeTimeSupplierMillis;
-            this.timeout = timeout;
-            this.startTimeMs = relativeTimeSupplierMillis.getAsLong();
-        }
-
-        @Override
-        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-            if (relativeTimeSupplierMillis.getAsLong() - startTimeMs > timeout.millis()) {
-                throw new ElasticsearchTimeoutException("Timed out getting mappings");
-            }
-
-            builder.startObject();
-            response.toXContent(builder, params);
-            builder.endObject();
-            return builder;
-        }
+            .getMappings(getMappingsRequest, new RestActionListener<>(channel) {
+                @Override
+                protected void processResponse(GetMappingsResponse getMappingsResponse) throws Exception {
+                    ensureOpen();
+                    channel.sendResponse(
+                        new RestResponse(
+                            RestStatus.OK,
+                            ChunkedRestResponseBody.fromXContent(
+                                () -> Iterators.concat(
+                                    Iterators.single((b, p) -> b.startObject()),
```

Comment:
I'm not quite sure why the `GetMappingsResponse` is a fragment and we had the complicated wrapping here but didn't want to change it in this PR so I manually wrapped it for now.